liblangutil/SourceReferenceFormatter.cpp

```diff
@@ -163,10 +173,15 @@ void SourceReferenceFormatter::printSourceLocation(SourceReference const& _ref)
 
 void SourceReferenceFormatter::printExceptionInformation(SourceReferenceExtractor::Message const& _msg)
 {
-	// exception header line
-	errorColored() << _msg.severity;
-	if (m_withErrorIds && _msg.errorId.has_value())
-		errorColored() << " (" << _msg.errorId.value().error << ")";
+    if (_msg.severity.eq("Error") && _msg.errorId.has_value())
+        errorColored() << " (" << _msg.errorId.value().error << ")";
+
+    else if(_msg.severity.eq("Warning") && _msg.errorId.has_value())
+        warningColored() << " (" << _msg.errorId.value().error << ")";
+
+    else if(_msg.severity.eq("Info") && _msg.errorId.has_value())
```

Comment:
@leonardoalt @axic Looks like we could use something like https://github.com/ethereum/solidity/pull/11606#discussion_r703078009 here. If we had a function a function to convert from `string` to `Severity`, we could use the enum here (with an `else` for unrecognized cases). Without it we have to hard-code strings here.

Do you have anything against adding a function like that to [`liblangutil/Exceptions.h`](https://github.com/ethereum/solidity/blob/develop/liblangutil/Exceptions.h) in this PR?